.github/workflows/gcp-deploy.yaml

```diff
@@ -48,158 +70,192 @@ jobs:
+    name: "Create New SQL User & Database"
     runs-on: ubuntu-latest
     needs: [env, delete-db]
 
     steps:
-      - name: Set Up gcloud CLI
+      - name: Set Up GCloud CLI
         uses: google-github-actions/setup-gcloud@v0.2.1
         with:
           project_id: ${{needs.env.outputs.PROJECT_ID}}
           service_account_key: ${{secrets.GCP_SA_KEY}}
           export_default_credentials: true
 
-      - name: Create User and Database (${{needs.env.outputs.DB_NAME}}) for Hasura
+      - name: "Create GCloud SQL User: ${{needs.env.outputs.DB_NAME}}"
         run: |
-          gcloud -q sql users create ${{needs.env.outputs.DB_NAME}} -i ${{needs.env.outputs.CLOUDSQL_INSTANCE_NAME}} --password ${{needs.env.outputs.DB_PASSWORD}}
-          gcloud -q sql databases create ${{needs.env.outputs.DB_NAME}} -i ${{needs.env.outputs.CLOUDSQL_INSTANCE_NAME}}
+          gcloud -q sql users create ${{needs.env.outputs.DB_NAME}} \
+            -i ${{needs.env.outputs.CLOUDSQL_INSTANCE_NAME}} \
+            --password ${{needs.env.outputs.DB_PASSWORD}}
+
+      - name: "Create Database: ${{needs.env.outputs.DB_NAME}}"
+        run: |
+          gcloud -q sql databases create ${{needs.env.outputs.DB_NAME}} \
+            -i ${{needs.env.outputs.CLOUDSQL_INSTANCE_NAME}}
 
   undeploy-backend:
     name: Undeploy Backend
     runs-on: ubuntu-latest
     needs: [env]
 
     steps:
-      - name: Set Up gcloud CLI
+      - name: Set Up GCloud CLI
         uses: google-github-actions/setup-gcloud@v0.2.1
         with:
           project_id: ${{needs.env.outputs.PROJECT_ID}}
           service_account_key: ${{secrets.GCP_SA_KEY}}
           export_default_credentials: true
 
-      - name: Undeploy Backend
+      - name: "Delete Service: ${{needs.env.outputs.BACKEND_SERVICE}}"
         continue-on-error: true
-        run: gcloud -q run services delete ${{needs.env.outputs.BACKEND_SERVICE}} --region ${{needs.env.outputs.CLOUDRUN_REGION}}
+        run: |
+          gcloud -q run services delete ${{needs.env.outputs.BACKEND_SERVICE}} \
+            --region ${{needs.env.outputs.CLOUDRUN_REGION}}
 
   delete-backend:
-    name: Remove Backend Image
+    name: Remove Backend Container Image
     runs-on: ubuntu-latest
     needs: [env]
 
     steps:
-      - name: Login to Registry
+      - name: "Login to Registry: ${{needs.env.outputs.DOCKER_REGISTRY}}"
         uses: docker/login-action@v1
         with:
           registry: ${{needs.env.outputs.DOCKER_REGISTRY}}
           username: _json_key
           password: ${{secrets.GCP_SA_KEY}}
 
-      - name: Set Up gcloud CLI
+      - name: Set Up GCloud CLI
         uses: google-github-actions/setup-gcloud@v0.2.1
         with:
           project_id: ${{needs.env.outputs.PROJECT_ID}}
           service_account_key: ${{secrets.GCP_SA_KEY}}
           export_default_credentials: true
 
-      - name: Delete Backend Image
+      - name: "Delete Container: ${{needs.env.outputs.BACKEND_TAG}}"
```

Comment:
i have recently learned the proper definitions of this terminology: a container image is a static file thats more or less just a snapshot of a container while a container is a running instance of a container image
you seem to mix those up in multiple places (i also did in the original workflow)